### PR TITLE
[SofaSphFluid] Fix ParticleSink init and refresh all examples.

### PR DESCRIPTION
**Pull request by epernod**
ParticleSink was not checking if the topology pointer used was valid before using it in the TopologyHandler.
Moreover init method was not checking if the topology is also dynamic which is a requirement to remove points.

A second check more general in TopologyHandler will be done in a second PR.

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by epernod** [comment]
[ci-build][with-all-tests]

**Comment by fredroy** [comment]
@epernod this PR fixes ParticleSink alright but two other tests fail _because_ of the new msg_error() in [ParticleSink.inl](https://github.com/sofa-framework/sofa/blob/aeea55fbd91e165ee323e3f4fcecb2e430135328/applications/plugins/SofaSphFluid/src/SofaSphFluid/ParticleSink.inl#L82) 😅

File: applications/plugins/SofaSphFluid/src/SofaSphFluid/ParticleSink.inl [hunk 1]
```diff
@@ -62,14 +65,30 @@ void ParticleSink<DataTypes>::init()
...
-    // Initialize functions and parameters for topology data and handler
-    d_fixed.createTopologyHandler(_topology);
-    d_fixed.registerTopologicalData();
+    if (_topology) // If topology is found, will check if it is dynamic. Otherwise only mechanical Object should be used.
+    {
+        _topology->getContext()->get(m_topoModifier);
+        if (!m_topoModifier)
+        {
+            sofa::core::objectmodel::BaseObject::d_componentState.setValue(sofa::core::objectmodel::ComponentState::Invalid);
+            msg_error() << "Topology has been found in at component: " << _topology->getName() << " but this topology is not dynamyc. ParticleSink will not be able to remove points.";
```
Comment (fredroy) [inline applications/plugins/SofaSphFluid/src/SofaSphFluid/ParticleSink.inl]: - Should it be a warning instead ?
- and typo (_dynamic_)

**Review by fredroy — CHANGES_REQUESTED** [review]

File: applications/plugins/SofaSphFluid/src/SofaSphFluid/ParticleSink.inl (hunk 1 above)
Comment (epernod) [reply applications/plugins/SofaSphFluid/src/SofaSphFluid/ParticleSink.inl]: component is set to invalid so no it should be an error.

Comment (fredroy) [reply applications/plugins/SofaSphFluid/src/SofaSphFluid/ParticleSink.inl]: Then Components/collision.SpatialGridPointModel.scn and Components/visualmodel.PointSplatModel.scn should be fixed (raising this msg_error)

Comment (epernod) [reply applications/plugins/SofaSphFluid/src/SofaSphFluid/ParticleSink.inl]: this means those scenes depends on the plugin?

**Comment by epernod** [comment]
[ci-build][with-all-tests]

**Pull request merged**